### PR TITLE
Plot recipe for MDS from MultivariateStats.jl

### PR DESCRIPTION
**Pull request by kescobo**
This is dependent on https://github.com/JuliaStats/MultivariateStats.jl/pull/85, though I'm not quite sure how to indicate that in compat until it's tagged. 

```
using StatsPlots
using Distances
using MultivariateStats

m = rand(100, 10)

dm = pairwise(BrayCurtis(), m)

mds = fit(MDS, dm, distances=true)

plot(mds)
```

mds

### CODE REVIEW

**Comment by mkborregaard** [comment]
That looks sweet! Have you considered an interface for multiple dimensions? Either passing dimensions manually, or using subplots or possibly 3d?

**Comment by kescobo** [comment]
@mkborregaard I have considered, but wasn't sure about the implementation. Since this is not a user recipe, can I add a keyword for that?

**Comment by mkborregaard** [comment]
Sure :-)

**Comment by mkborregaard** [comment]
(technically this is a user recipe)

**Comment by kescobo** [comment]
I think I was mis-remembering the deal with keyword arguments... 

In any case, this now allows a user to specify axes, and do a 3rd axis:

```julia
using StatsPlots
using Distances
using MultivariateStats

m = rand(100, 10)
dm = pairwise(BrayCurtis(), m)
mds = fit(MDS, dm, distances=true)

plot(mds)
```

mds1

```julia
plot(mds, mds_axes=(3,5))
```

mds2


```julia
plot(mds, mds_axes=(1,2,3))
```

mds3

**Comment by mkborregaard** [comment]
Love it :heart:

**Comment by mkborregaard** [comment]
Oh - I merged already, but only now realised - shouldn't `aspect_ratio` have been set to 1?

File: src/mds_scatter.jl [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@recipe function f(mds::MDS{<:Real}; mds_axes=(1,2))
```
Comment (mkborregaard) [inline src/mds_scatter.jl]: Also, I think you're missing a dep on MultivariateStats for this to work?

Comment (mkborregaard) [reply src/mds_scatter.jl]: Wait, you do have that, I seem to have a local problem, sorry for the noise

**Comment by kescobo** [comment]
Not sure, what does that do? Happy to add it in.

**Comment by mkborregaard** [comment]
@kescobo I'm getting `WARNING: could not import MultivariateStats.MDS into Main` locally - is there something wrong with my setup/versions?

Also, could the signature be wider to allow for PCAs and other ordinations as well? Is the interface not general enough for this?

It would be awesome to have a readme example with this code
```julia
using MultivariateStats, RDatasets, StatsPlots

iris = dataset("datasets", "iris")
Xtr = convert(Matrix, iris[:, 1:4])'
M = fit(PCA, Xtr; maxoutdim=2)
plot(M, group = iris.Species)
```
or something like that. Compare the code here: https://multivariatestatsjl.readthedocs.io/en/latest/pca.html#data-analysis

**Comment by mkborregaard** [comment]
> Not sure, what does that do?

Forces the axes to have the same scale. Given that they are by definition on the same scale it gives a more correct representation of the spread on the axes.

**Comment by kescobo** [comment]
>@kescobo I'm getting WARNING: could not import MultivariateStats.MDS into StatsPlots locally - is there something wrong with my setup/versions?

You have to be on master of MultivariatStats (see the first comment in this thread=). The PR introducing this API hasn't been tagged, so I can't add it to `[compat]` in the `Project.toml` yet (I asked on slack)

>Also, could the signature be wider to allow for PCAs and other ordinations as well? Is the interface not general enough for this?

I haven't looked into it, but I thought about the same thing. I will happily do a separate PR.

>It would be awesome to have a readme example with this code

Sure - should this be a separate PR since this one is already merged?

**Comment by kescobo** [comment]
> Forces the axes to have the same scale. Given that they are by definition on the same scale. Given that they are by definition on the same scale it gives a more correct representation of the spread on the axes.

My understanding was that for PCoA at least, the scales are essentially unitless and so scaling is meaningless. Entirely possible I misunderstood that though 😆

**Comment by mkborregaard** [comment]
> You have to be on master of MultivariatStats (see the first comment in this thread=)

Sorry :joy:

> I will happily do a separate PR.

Yay! 💖 

> should this be a separate PR

I think it has to be yeah.

> scales are essentially unitless 

AFAIK the PCoA transformation maintains distances among points. But if Plots chooses different scales for the x and y axis (based on the range of the data and the shape of the plotting window), the distances will not be correct.

**Comment by kescobo** [comment]
>AFAIK the PCoA transformation maintains distances among points.

But PCoA is for non-metric distances right? Maybe I'll ask on Slack...

**Pull request merged**